api/v1alpha4/machineset_types.go

```diff
@@ -36,6 +36,7 @@ type MachineSetSpec struct {
 	// This is a pointer to distinguish between explicit zero and unspecified.
 	// Defaults to 1.
 	// +optional
+	// +kubebuilder:default=1
 	Replicas *int32 `json:"replicas,omitempty"`
```

Comment:
Does this allow us to specify Replicas = 0?

Reply:
Yes it does, we use this in openshift already and you can have any valid value that is an integer. This only gets defaulted when there is no replicas field at all (eg if you patch to delete the replicas field it would be re-added by openAPI and set to value 1, or you omit it when creating the resource)